### PR TITLE
Add endpoint for getting combiner quota

### LINKED ISSUE #9716: Combiner PNP quota endpoint
For ASv2 and the new version of the rate limit defined in https://www.notion.so/clabsco/ODIS-Rate-Limit-v2-2d3e7bcf01cf4fd19f00b1542ae862b9, clients will need to be able to easily check their remaining ODIS quota and top up if necessary. 

Because Signers can have different views of an account's performed quota count and (consequently) remaining ODIS quota, the combiner should be able to aggregate quota responses from all the signers into a single summary response.

Ex. For an ODIS instance with a 2/3 threshold, where signer[i] believes the user has i remaining quota, the user has 1 remaining quota. This is because signer[0] will not accept any requests (believing the user has zero quota), while signer[1] and signer[2] will both accept one request, but not two. This means the user can make at most one request where a threshold of signers will accept. 

In general, the combiner should return the "nth least restrictive" (or technically n-1th least restrictive) view of the user's quota for a system with threshold n/m. See the code in https://github.com/celo-org/celo-monorepo/blob/794ab6b0aa7afbe6479f9d9357f93dea05854f25/packages/phone-number-privacy/combiner/src/domain/services/thresholdState.ts#L10 to get a sense of how this is done for the domains use case.

### PR DESCRIPTION
**Pull request by isabellewei**
### Description

Adds new endpoint in the combiner to check the quota status for a given account

### Tested

Not yet

### Related issues

- Fixes #9716

### CODE REVIEW

File: packages/phone-number-privacy/combiner/src/pnp/services/thresholdState.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  findCombinerQuotaState(session: Session<R>) {
+    const signerResponses: PnpQuotaResponseSuccess[] = session.responses
+      .map((signerResponse) => signerResponse.res)
+      .filter((res) => res.success)
+      .sort((a, b) => a.totalQuota - b.totalQuota)
+
+    const threshold = this.config.keys.threshold
+    if (signerResponses.length < threshold) {
+      throw new Error('Insufficient number of successful signer responses')
```
Comment (alecps) [inline packages/phone-number-privacy/combiner/src/pnp/services/thresholdState.ts:16]: TODO(Alec) check error handling

File: packages/phone-number-privacy/combiner/src/pnp/services/thresholdState.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { OdisConfig } from '../../config'
+
+export class CombinerThresholdStateService<R extends PnpQuotaRequest> {
+  constructor(readonly config: OdisConfig) {}
+
+  findCombinerQuotaState(session: Session<R>) {
+    const signerResponses: PnpQuotaResponseSuccess[] = session.responses
+      .map((signerResponse) => signerResponse.res)
+      .filter((res) => res.success)
+      .sort((a, b) => a.totalQuota - b.totalQuota)
```
Comment (alecps) [inline packages/phone-number-privacy/combiner/src/pnp/services/thresholdState.ts]:
```suggestion
      .sort((a, b) => (a.totalQuota - a.performedQueryCount) - (b.totalQuota - b.performedQueryCount)
```
The way this works and the naming is a bit confusing :/ feel free to suggest improvements!

File: packages/phone-number-privacy/combiner/src/pnp/services/thresholdState.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+    const threshold = this.config.keys.threshold
+    if (signerResponses.length < threshold) {
+      throw new Error('Insufficient number of successful signer responses')
+    }
+
+    return signerResponses[threshold].totalQuota
```
Comment (alecps) [inline packages/phone-number-privacy/combiner/src/pnp/services/thresholdState.ts]: Should this be [threshold - 1]?

File: packages/phone-number-privacy/common/src/interfaces/errors.ts [hunk 4]
```diff
@@ -22,6 +22,7 @@ export enum ErrorMessage {
   INVALID_KEY_VERSION_RESPONSE = `CELO_ODIS_ERR_20 SIG_ERR Signer response key version header is invalid`,
   INVALID_SIGNER_RESPONSE = `CELO_ODIS_ERR_21 SIG_ERR Signer response body is invalid`,
   SIGNER_RESPONSE_FAILED_WITH_OK_STATUS = `CELO_ODIS_ERR_22 SIG_ERR Signer response failed with 200 status`,
+  COMBINER_QUOTA_STATUS_FAILURE = `CELO_ODIS_ERR_23 SIG_ERR Failed to get quota status from a threshold of signatures`,
```
Comment (alecps) [inline packages/phone-number-privacy/common/src/interfaces/errors.ts]: To follow the pattern used above consider 
```suggestion
  THRESHOLD_PNP_QUOTA_STATUS_FAILURE = `CELO_ODIS_ERR_23 SIG_ERR Failed to get PNP quota status from a threshold of signers`,,
```

File: packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/action.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      try {
+        const quotaStatus = this.thresholdStateService.findCombinerQuotaState(session)
+        this.io.sendSuccess(200, session.response, quotaStatus)
+        return
+      } catch (error) {
+        session.logger.error({ error }, 'Error combining signer quota status responses')
+      }
+    }
+    this.io.sendFailure(
+      ErrorMessage.THRESHOLD_DOMAIN_QUOTA_STATUS_FAILURE,
```
Comment (alecps) [inline packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/action.ts]:
```suggestion
      ErrorMessage.THRESHOLD_PNP_QUOTA_STATUS_FAILURE,
```

File: packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/action.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import { IO } from '../../../common/io'
+import { Session } from '../../../common/session'
+import { OdisConfig } from '../../../config'
+import { CombinerThresholdStateService } from '../../services/thresholdState'
+
+export class PnPQuotaAction extends CombineAction<PnPQuotaRequest> {
+  constructor(
+    readonly config: OdisConfig,
+    readonly thresholdStateService: CombinerThresholdStateService<PnPQuotaRequest>,
+    readonly io: IO<PnPQuotaRequest>
```
Comment (alecps) [inline packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/action.ts]: This is probably something I could've made clearer with naming. The IO class is abstract, and we use its abstract functions in this file. That means we need to implement an IO class for this endpoint that extends the abstract class  and use that here. I can help you with that in a quick 1:1!

**Review by alecps — CHANGES_REQUESTED** [review]
Looks great!! Just a couple small comments

File: packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/action.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import { ErrorMessage, PnPQuotaRequest } from '@celo/phone-number-privacy-common'
+import { CombineAction } from '../../../common/combine'
+import { IO } from '../../../common/io'
+import { Session } from '../../../common/session'
+import { OdisConfig } from '../../../config'
+import { CombinerThresholdStateService } from '../../services/thresholdState'
+
+export class PnPQuotaAction extends CombineAction<PnPQuotaRequest> {
```
Comment (eelanagaraj) [inline packages/phone-number-privacy/combiner/src/pnp/endpoints/quota/action.ts]: nit: `PnP` -> `Pnp` in the newly added classes/types/etc. to be consistent with existing naming

**Review by alecps — APPROVED** [review]
Looks awesome!!!

**Pull request merged**